calculator_test.go

```diff
@@ -2,23 +2,302 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	testCases := []struct {
+		name  string
+		a, b  float64
+		extra []float64
+		want  float64
+	}{
+		{
+			name:  "Sum of 2 operands to give a positive number",
+			a:     1,
+			b:     1,
+			extra: nil,
+			want:  2,
+		},
+		{
+			name:  "Sum of 3 positive numbers to give a positive number",
+			a:     1,
+			b:     1,
+			extra: []float64{1},
+			want:  3,
+		},
+		{
+			name:  "Sum of 5 negative numbers to give a negative number",
+			a:     -2,
+			b:     -2,
+			extra: []float64{-2, -2, -2},
+			want:  -10,
+		},
+		{
+			name:  "Sum of 4 numbers to give 0 sum",
+			a:     -1,
+			b:     -1,
+			extra: []float64{1, 1},
+			want:  0,
+		},
+	}
+	for _, tc := range testCases {
+		got := calculator.Add(tc.a, tc.b, tc.extra...)
+		if tc.want != got {
+			t.Errorf("%s: Add(%v, %v, %+v) want %f, got %f",
+				tc.name, tc.a, tc.b, tc.extra, tc.want, got)
+		}
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+func TestSubtract(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name  string
+		a, b  float64
+		extra []float64
+		want  float64
+	}{
+		{
+			name:  "Difference of 2 equal positive numbers to give 0",
+			a:     1,
+			b:     1,
+			extra: nil,
+			want:  0,
+		},
+		{
+			name:  "Difference of 2 equal negative numbers to give 0",
+			a:     -1,
+			b:     -1,
+			extra: nil,
+			want:  0,
+		},
+		{
+			name:  "Difference of more than 2 arguments",
+			a:     10,
+			b:     1,
+			extra: []float64{1, 1, 1, 1, 1, 1, 1, 1, 1},
+			want:  0,
+		},
+	}
+	for _, tc := range testCases {
+		got := calculator.Subtract(tc.a, tc.b, tc.extra...)
+		if tc.want != got {
+			t.Errorf("%s: Subtract(%v, %v, %+v)) want %f, got %f",
+				tc.name, tc.a, tc.b, tc.extra, tc.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
```

Comment:
Apart from the randomness, we could really test all three of Add, Subtract, and Multiply in the same test, couldn't we?

Reply:
Yes! This change helped improve the DRYness of the code.